test/plug/conn_test.exs

```diff
@@ -837,4 +837,22 @@ defmodule Plug.ConnTest do
       conn |> register_before_send(fn(_) -> nil end)
     end
   end
+
+  test "does not delegate to connections' adapter's chunk/2 when called with emtpy chunk" do
+    defmodule RaisesOnEmptyChunkAdapter do
```

Comment:
Having no experience with purging a module, would `:code.purge(RaisesOnEmptyChunkAdapter)` be sufficient, or would `:code.delete/1` also be necessary? Is this just to be sure that we're not using code defined anywhere else?

Something like this at the end of the test?
```
    ...
    :code.purge(RaisesOnEmptyChunkAdapter)
    :code.delete(RaisesOnEmptyChunkAdapter)
  end
```

If this is too smelly for tests, I'm also open to other methods of asserting that the underlying adapter's `chunk/2` is not being called, if anyone has suggestions?